mitodl/teachersportal
# Upgraded NPM dependencies

#### What are the relevant tickets?

Related to #476 
#### What's this PR do?

Upgrades dependencies and removes unused ones. In order to fix compatibility issues certain changes were made:
- shopping cart is now opened using a prop instead of our hacky workaround
- `ReviewsTab` is removed, which wasn't used anyway
- `onEnterKeyDown` was removed in LoginModal and replaced with `onKeyDown` according to the warning's recommendation
- background-color is marked `!important` to fix the blue background issue which is the original reason we haven't upgraded material-ui
- tests are updated to click the material-ui button properly
#### Where should the reviewer start?

package.json
#### How should this be manually tested?

The reviewer should go through the checkout workflow and make sure everything looks right. In addition they should also use the enter key on every field in the login modal to confirm that it still works as expected. (This is not covered by tests unfortunately.)
#### Any background context you want to provide?

This also pins dependencies to exact versions to avoid issues in the future.
#### Screenshots (if appropriate)
#### What GIF best describes this PR or how it makes you feel?
